.circleci/config.yml

```diff
@@ -0,0 +1,61 @@
+version: 2
+jobs:
+  test_py:
+    machine:
+      image: circleci/classic:201710-01
+    steps:
+      - checkout
+      - run:
+          name: Git Submodule
+          command: |
+            git submodule sync
+            git submodule update --init --recursive
+            echo "$CIRCLE_SHA1" > .current-git-hash
+      - restore_cache:
+          keys:
+            - v1-docker-layers-{{ checksum ".current-git-hash" }}
+            - v1-docker-layers-
```

Comment:
when does this key get saved?

Reply:
found the answer in https://circleci.com/docs/2.0/caching/#restoring-cache

Reply:
I agree it's odd, but the system does seem like a good one. It's because once a cache is saved its immutable, so you save a very specific key but then you can restore more loosely. I kinda like it actually. It'd be better if we could use their "docker layer caching" but it's a premium paid add-on and we'll hopefully soon be done with Circle :)